logspec-worker/logspec_api.py

```diff
@@ -219,15 +242,27 @@ def generate_issues_and_incidents(result_id, log_file, test_type, origin):
     parser = test_types[test_type]["parser"]
     error_list, new_status = process_log(log_file, parser, start_state)
     for error in error_list:
+        issue_id = ""
+        issue_version = None
         if error and error["error"].get("signature"):
             # do not generate issues for error_return_code since they are not
             # fatal, avoid noise.
             if error["error"].get("error_type") == "linux.kernel.error_return_code":
                 continue
 
+            if test_type == 'build':
+                # Use logspec signature_loc to fold under the same issue
+                # build failures reported by different compilers
+                signature_loc = error["error"].get("signature_loc")
+                issue = look_up_existing_issue(cursor, origin, signature_loc)
+                if issue:
+                    issue_id = issue[0]
+                    issue_version = issue[1]
```

Comment:
We aren't covering `parsed_data["issue_node"].append(issue)` with this code.

How about we move `look_up_existing_issue` into `new_issue`, just to keep it clear that `new_issue` is the function wrapping anything related to converting logspec data into KCIDB data.
And inside of it we handle the id and version cleanly.

Reply:
This makes sense. Done.